server/ai_live_video.go

```diff
@@ -338,24 +377,29 @@ func startControlPublish(ctx context.Context, control *url.URL, params aiRequest
 		})
 	}
 
-	if control, exists := params.node.LivePipelines[stream]; exists {
+	if control, exists := params.node.LivePipelines[stream]; exists && control.ControlPub != nil && control.StopControl != nil {
 		clog.Info(ctx, "Stopping existing control loop", "existing_request_id", control.RequestID)
 		control.ControlPub.Close()
 		// TODO better solution than allowing existing streams to stomp over one another
 	}
 
-	params.node.LivePipelines[stream] = &core.LivePipeline{
-		ControlPub:  controlPub,
-		StopControl: stop,
-		RequestID:   params.liveParams.requestID,
+	if params.node.LivePipelines[stream] != nil {
+		params.node.LivePipelines[stream].ControlPub = controlPub
+		params.node.LivePipelines[stream].StopControl = stop
+		params.node.LivePipelines[stream].RequestID = params.liveParams.requestID
+	} else {
+		stopProcessing(ctx, params, fmt.Errorf("no control publisher found for stream %s", stream))
```

Comment:
I am a little confused, with the `!= nil` check above, when is `LivePipelines` actually populated?

Reply:
https://github.com/livepeer/go-livepeer/pull/3609/commits/55f06103cd2e3be2e7332f2fca015a5031ea6362